src/PolykeyClient.ts

```diff
@@ -34,51 +50,66 @@ interface PolykeyClient extends CreateDestroyStartStop {}
 )
 class PolykeyClient {
   static async createPolykeyClient({
-    nodePath = config.defaultsUser.nodePath,
-    streamFactory,
-    streamKeepAliveTimeoutTime,
-    parserBufferByteLimit,
+    port,
+    host,
+    options = {},
     fs = require('fs'),
     logger = new Logger(this.name),
     fresh = false,
   }: {
-    nodePath?: string;
-    streamFactory: StreamFactory;
-    streamKeepAliveTimeoutTime?: number;
-    parserBufferByteLimit?: number;
+    port: number;
+    host: string;
+    options: DeepPartial<PolykeyClientOptions>;
     fs?: FileSystem;
     logger?: Logger;
     fresh?: boolean;
   }): Promise<PolykeyClient> {
     logger.info(`Creating ${this.name}`);
-    if (nodePath == null) {
+    const optionsDefaulted = utils.mergeObjects(options, {
+      nodePath: config.defaultsUser.nodePath,
+      tokenBase: config.paths.tokenBase,
+      callTimeoutTime: config.defaultsSystem.rpcCallTimeoutTime,
+      parserBufferSize: config.defaultsSystem.rpcParserBufferSize,
+    });
+
+    if (optionsDefaulted.nodePath == null) {
       throw new errors.ErrorUtilsNodePath();
     }
-    await utils.mkdirExists(fs, nodePath);
-    const sessionTokenPath = path.join(nodePath, config.paths.tokenBase);
+    await utils.mkdirExists(fs, optionsDefaulted.nodePath);
+    const sessionTokenPath = path.join(
+      optionsDefaulted.nodePath,
+      optionsDefaulted.tokenBase,
+    );
     const session = await Session.createSession({
       sessionTokenPath,
       logger: logger.getChild(Session.name),
       fresh,
     });
+    const ws = await WebSocketClient.createWebSocketClient({
+      host: host,
+      port: port,
+      config: optionsDefaulted.wsConfig,
```

Comment:
I recall @CMCDragonkai mentioning this. We don't want to pass in the whole `wsConfig`. The `PolykeyClient` needs to provide some of these options without allowing the user to set them via the parameters. The main one is the custom validation.

The `NodeConnectionManger` has a pretty clear example of this when creating the `QUICServer`.

Reply:
uh idk how to go about this other than optionsDefaulted, wherein I have set it to false, if you have any other more concrete solutions, feel free to share

Reply:
We're going to use flattened options for EVERYTHING except PKAgent and PKClient. So I'm reverting some old changes. Monitor this @addievo I'll need you to take over after I'm finished to update all test call signatures.